.github/workflows/Build-Test-And-Deploy.yml

```diff
@@ -0,0 +1,88 @@
+name: Build, Test, and Deploy EssentialCSharp
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up .NET Core
+        uses: actions/setup-dotnet@v3
+        with:
+          global-json-file: global.json
+          source-url: https://pkgs.dev.azure.com/intelliTect/_packaging/EssentialCSharp/nuget/v3/index.json
+        env:
+          NUGET_AUTH_TOKEN: ${{secrets.AZURE_DEVOPS_PAT}}
+
+      - name: Set up dependency caching for faster builds
+        uses: actions/cache@v3
+        id: nuget-cache
+        with:
+          path: |
+            ~/.nuget/packages
+            ${{ github.workspace }}/**/obj/project.assets.json
+          key: ${{ runner.os }}-nuget-${{ hashFiles('**/packages.lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-nuget-${{ hashFiles('**/packages.lock.json') }}
+            ${{ runner.os }}-nuget-
```

Comment:
Do we need both of these lines? I would have expected the first to cover it.

Reply:
It's so if it doesn't match on the first cache, we will grab the second cache (which might not have all the packages we need, but has some), so it will still be a speedup than grabbing none. But we want it to grab the exact one if we can.

https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key